common.go

```diff
@@ -442,3 +449,17 @@ func GetLanguageByAlias(alias string) (lang string, ok bool) {
 
 	return
 }
+
+// GetLanguagesByGitAttributes returns either a string slice with the lenguage if the filename match with a regExp in .gitattributes
+//or return nil in case of none regexp matchs the filename . It complies with the signature to be a Strategy type.
+func GetLanguagesByGitAttributes(filename string, content []byte, candidates []string) []string {
+	if loadedGitAttributes != nil {
```

Comment:
The idiomatic way to do this in Go is to return nil if the loadedGitAttributes is == nil.  That way, the happy path of the function (the for loop) has the smallest level of indentation.  For a better explanation of this concept, see Golang UK Conference 2016 - Mat Ryer - Idiomatic Go Tricks (https://www.youtube.com/watch?v=yeetIgNeIkc).